tests/space-time-stack/CMakeLists.txt

```
@@ -1,18 +1,10 @@
 # A function to add such "simple" tests in 'tests/CMakeLists.txt' might be a good option.
-add_executable(test_space_time_stack_demangling)
-target_sources(
-    test_space_time_stack_demangling
-    PRIVATE
-        test_demangling.cpp
+kp_add_executable_and_test(
+    TARGET_NAME test_space_time_stack_demangling
+    SOURCE_FILE test_demangling
```

Why doesn't this need to be
```suggestion
    SOURCE_FILE test_demangling.cpp
```
?

---

I've added `.cpp`. But for posterity, it seems `CMake` will try to find your file by testing several suffixes:
```bash
-- Configuring done (1.1s)
CMake Error at tests/CMakeLists.txt:22 (target_sources):
  Cannot find source file:

    test_demanglig

  Tried extensions .c .C .c++ .cc .cpp .cxx .cu .mpp .m .M .mm .ixx .cppm
  .ccm .cxxm .c++m .h .hh .h++ .hm .hpp .hxx .in .txx .f .F .for .f77 .f90
  .f95 .f03 .hip .ispc
Call Stack (most recent call first):
  tests/space-time-stack/CMakeLists.txt:2 (kp_add_executable_and_test)


-- Generating done (0.0s)
```